src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -51,6 +51,18 @@ const DomainFragment = ({
       following: item.following !== undefined ? item.following : item.user_id_follower !== null
     }));
   }, [discovery.initialDomains]);
+  const newMapFollowedDomain = React.useMemo(() => {
+    return followedDomains.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [followedDomains]);
+  const newMapUnfollowedDomain = React.useMemo(() => {
+    return unfollowedDomains.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [unfollowedDomains]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useMemo` hooks for `newMapFollowedDomain` and `newMapUnfollowedDomain` seem to be doing the same operation on different arrays. This is a repetition of logic which can be avoided by creating a separate function that performs this operation and then using it inside the `useMemo` hooks.

```diff
+  const mapDomains = (domains) => {
+    return domains.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  };

-  const newMapFollowedDomain = React.useMemo(() => {
-    return followedDomains.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
-  }, [followedDomains]);
-  const newMapUnfollowedDomain = React.useMemo(() => {
-    return unfollowedDomains.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
-  }, [unfollowedDomains]);

+  const newMapFollowedDomain = React.useMemo(() => mapDomains(followedDomains), [followedDomains]);
+  const newMapUnfollowedDomain = React.useMemo(() => mapDomains(unfollowedDomains), [unfollowedDomains]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->